src/android/jni/utils.cpp

```diff
@@ -0,0 +1,173 @@
+
+    // preparing struct to be filled by API function
+    float* pixel = new float[2];
+    pixel[0] = 0.f;
+    pixel[1] = 0.f;
+
+    rs2_project_point_to_pixel(pixel, &intrinsics, point);
+
+    // returning pixel into pixel_2D struct
+    jclass pixel_2D_class = env->GetObjectClass(pixel_2D);
+    jfieldID pixel_x_field = env->GetFieldID(pixel_2D_class, "mX", "I");
+    jfieldID pixel_y_field = env->GetFieldID(pixel_2D_class, "mY", "I");
+    env->SetIntField(pixel_2D, pixel_x_field, pixel[0]);
+    env->SetIntField(pixel_2D, pixel_y_field, pixel[1]);
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_com_intel_realsense_librealsense_Utils_nDeprojectPixelToPoint(JNIEnv *env, jclass type,
+                                                                 jobject point_3D, jobject intrinsic,
+                                                                 jobject pixel_2D, jfloat depth) {
+
+    // retrieving float[2] from pixel_2D
+    jclass pixel_2D_class = env->GetObjectClass(pixel_2D);
+    jfieldID pixel_x_field = env->GetFieldID(pixel_2D_class, "mX", "I");
+    jfieldID pixel_y_field = env->GetFieldID(pixel_2D_class, "mY", "I");
+    float* pixel = new float[2];
+    pixel[0] = env->GetIntField(pixel_2D, pixel_x_field);
+    pixel[1] = env->GetIntField(pixel_2D, pixel_y_field);
+
+    // retrieving rs2_intrinsics from intrinsic object
+    jclass intrinsic_class = env->GetObjectClass(intrinsic);
+    jfieldID width_field = env->GetFieldID(intrinsic_class, "mWidth", "I");
+    jfieldID height_field = env->GetFieldID(intrinsic_class, "mHeight", "I");
+    jfieldID ppx_field = env->GetFieldID(intrinsic_class, "mPpx", "F");
+    jfieldID ppy_field = env->GetFieldID(intrinsic_class, "mPpy", "F");
+    jfieldID fx_field = env->GetFieldID(intrinsic_class, "mFx", "F");
+    jfieldID fy_field = env->GetFieldID(intrinsic_class, "mFy", "F");
+    jfieldID model_field = env->GetFieldID(intrinsic_class, "mModelValue", "I");
+    jfieldID coeff_field = env->GetFieldID(intrinsic_class, "mCoeffs", "[F");
+    rs2_intrinsics intrinsics;
+    intrinsics.width = env->GetIntField(intrinsic, width_field);
+    intrinsics.height = env->GetIntField(intrinsic, height_field);
+    intrinsics.ppx = env->GetFloatField(intrinsic, ppx_field);
+    intrinsics.ppy = env->GetFloatField(intrinsic, ppy_field);
+    intrinsics.fx = env->GetFloatField(intrinsic, fx_field);
+    intrinsics.fy = env->GetFloatField(intrinsic, fy_field);
+    intrinsics.model = (rs2_distortion)env->GetIntField(intrinsic, model_field);
+
+    jobject coeffsObject = env->GetObjectField(intrinsic, coeff_field);
+    jfloatArray* coeffsArray = reinterpret_cast<jfloatArray *>(&coeffsObject);
+    jfloat * coeffs = env->GetFloatArrayElements(*coeffsArray, NULL);
+    memcpy(intrinsics.coeffs, coeffs, 5 * sizeof(float));
+    env->ReleaseFloatArrayElements(*coeffsArray, coeffs, 0);
+
+    // preparing struct to be filled by API function
+    float* point = new float[3];
+    point[0] = 0.f;
```

Comment:
Redundant - new  initializes the values. But probably you don't have to use "new" here

Reply:
done